### PR TITLE
[BEAM-4215]Pick up the latest googledatastore release

### PR DESCRIPTION
**Pull request by liyanhui1228**
Update the googledatastore dependency upper bound to allow picking up the upcoming [7.1.0](https://github.com/GoogleCloudPlatform/google-cloud-datastore/pull/183) release.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/python/setup.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def get_version():
     # google-apitools 0.5.23 and above has important Python 3 supports.
     'google-apitools>=0.5.23,<=0.5.24',
     'proto-google-cloud-datastore-v1>=0.90.0,<=0.90.4',
-    'googledatastore>=7.0.1,<=7.1.0; python_version < "3.0"',
+    'googledatastore>=7.0.1,<7.2.0; python_version < "3.0"',
```
Comment (tvalentyn) [inline sdks/python/setup.py]: Shall we increase the lower bound?

**Comment by aaltay** [comment]
Datastore tests are failing

14:52:39 Traceback (most recent call last):
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
14:52:39     addr.filename, addr.module)
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
14:52:39     return self.importFromDir(dir_path, fqname)
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
14:52:39     mod = load_module(part_fqname, fh, filename, desc)
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/apache_beam/io/gcp/datastore_write_it_test.py", line 38, in <module>
14:52:39     from apache_beam.io.gcp import datastore_write_it_pipeline
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/apache_beam/io/gcp/datastore_write_it_pipeline.py", line 38, in <module>
14:52:39     from apache_beam.io.gcp.datastore.v1.datastoreio import DeleteFromDatastore
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/apache_beam/io/gcp/datastore/v1/datastoreio.py", line 27, in <module>
14:52:39     from apache_beam.io.gcp.datastore.v1 import helper
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/apache_beam/io/gcp/datastore/v1/helper.py", line 46, in <module>
14:52:39     from googledatastore import PropertyFilter, CompositeFilter
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/googledatastore/__init__.py", line 21, in <module>
14:52:39     from . import helper
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/googledatastore/helper.py", line 25, in <module>
14:52:39     from google.datastore.v1 import entity_pb2
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/google/datastore/v1/entity_pb2.py", line 28, in <module>
14:52:39     dependencies=[google_dot_api_dot_annotations__pb2.DESCRIPTOR,google_dot_protobuf_dot_struct__pb2.DESCRIPTOR,google_dot_protobuf_dot_timestamp__pb2.DESCRIPTOR,google_dot_type_dot_latlng__pb2.DESCRIPTOR,])
14:52:39   File "/home/jenkins/jenkins-slave/workspace/beam_PreCommit_Python_Commit/src/sdks/python/target/.tox/py27-gcp/local/lib/python2.7/site-packages/google/protobuf/descriptor.py", line 878, in __new__
14:52:39     return _message.default_pool.AddSerializedFile(serialized_pb)
14:52:39 TypeError: Couldn't build proto file into descriptor pool!
14:52:39 Invalid proto descriptor for file "google/datastore/v1/entity.proto":

**Comment by tvalentyn** [comment]
Some Datastore tests are failing, you can see failing tests in https://builds.apache.org/job/beam_PreCommit_Python_Commit/2637/consoleFull

**Comment by tvalentyn** [comment]
You can search for "14:52:39 ERROR: Failure: TypeError (Couldn't build proto file into descriptor pool!"

**Comment by tvalentyn** [comment]
(didn't see @aaltay's comment).

**Comment by liyanhui1228** [comment]
Looking into googledatastore, the latest version changed the proto dependency from `proto-google-cloud-datastore-v1` to `proto-google-datastore-v1`, which is causing the test to fail. Both the two proto packages should be deprecated and no one should depend on them directly. I'm talking to them to see whether googledatastore can change the proto dependency back to the previous one and do another release. The new proto package also introduced new version conflicts, which makes the problem worse than before. I'll rerun the test when I get back from googledatastore.

**Comment by liyanhui1228** [comment]
googledatastore rollback their change for the proto dependency, and did a new release(version 7.0.2) which only contains the `httplib2` version update, all the rest remains the same as the current version beam is depending on. As this release is version 7.0.2 which falls between the current version range in beam, this PR is no longer needed. I'm closing this...

**Comment by aaltay** [comment]
Thank you for resolving this and tracking the long term solution.

**Pull request closed without merging**